audiostream2py/data.py

```diff
@@ -0,0 +1,148 @@
+        return AudioData(start_date, end_date, waveform, frame_count, status_flags)
+
+    def __lt__(self, other: Union['AudioData', int, float]) -> bool:
+        """Less than comparing start_date
+
+        :param other: AudioData | int | float
+        :return: bool
+        """
+        other_val = other.start_date if isinstance(other, AudioData) else other
+        return self.start_date < other_val
+
+    def __gt__(self, other: Union['AudioData', int, float]) -> bool:
+        """Greater than comparing start_date
+
+        :param other: AudioData | int | float
+        :return: bool
+        """
+        other_val = other.start_date if isinstance(other, AudioData) else other
+        return self.start_date > other_val
+
+    def __getitem__(self, index) -> 'AudioData':
+        """Slice waveform and return AudioData with updated timestamps and data.
+
+        Further info on methodology found in timestamping data stream discussion:
+        https://miro.com/app/board/uXjVPsuJtdM=/?share_link_id=872583858418
+
+        Note: slice 'step' is unlikely used in the traditional sense which is how it is implemented.
+        A future improvement could use 'step' as a channel selector by filtering out bytes of other
+        channels.
+
+        :param index: timestamp or slice
+        :return: Sliced AudioData
+        """
+        if isinstance(index, slice):
+            if index.start is not None:
+                start_sample, start_date = self.nearest_sample_index_and_time(
+                    timestamp=index.start, rounding_type='ceil'
+                )
+            else:
+                start_sample, start_date = 0, self.start_date
+
+            if index.stop is not None:
+                end_sample, end_date = self.nearest_sample_index_and_time(
+                    timestamp=index.stop, rounding_type='floor'
+                )
+            else:
+                end_sample, end_date = self.frame_count, self.end_date
+
+            start = start_sample * self.frame_size
+            stop = end_sample * self.frame_size
+            step = index.step * self.frame_size if index.step is not None else None
+            waveform = self.waveform[start:stop:step]
+            frame_count = end_sample - start_sample
+        else:
+            sample_idx, start_date = self.nearest_sample_index_and_time(
+                timestamp=index, rounding_type='floor'
+            )
+            _, end_date = self.nearest_sample_index_and_time(
+                timestamp=index, rounding_type='ceil'
+            )
+            waveform = self.waveform[sample_idx : sample_idx + self.frame_size]
+            frame_count = 1
+
+        return AudioData(start_date, end_date, waveform, frame_count, self.status_flags)
+
+    @cached_property
+    def frame_size(self) -> int:
+        """Byte count per frame"""
+        return int(len(self.waveform) / self.frame_count)
+
+    def nearest_sample_index_and_time(
+        self, timestamp, rounding_type='ceil'
```

Comment:
I would do `rounding_type: Literal['floor', 'ceil'] = 'ceil'` and validate that the value is valid.